modules/core/src/effects/post-process-effect.js

```diff
@@ -0,0 +1,123 @@
+import Effect from '../lib/effect';
+import ScreenPass from '../passes/screen-pass';
+/* eslint-disable import/no-extraneous-dependencies */
+import {normalizeShaderModule} from '@luma.gl/shadertools';
+import {ClipSpace} from '@luma.gl/core';
+
+export default class PostProcessEffect extends Effect {
+  constructor(module, props = {}) {
+    super(props);
+    this.id = `${module.name}-pass`;
+    normalizeShaderModule(module);
+    this.module = module;
+  }
+
+  prepare(gl) {
+    if (!this.passes) {
+      this.passes = normalizePasses(gl, this.module, this.id, this.props);
+    }
+  }
+
+  render(gl, params) {
+    const {target = null} = params;
+    let switchBuffer = false;
+    for (let index = 0; index < this.passes.length; index++) {
+      const inputBuffer = switchBuffer ? params.outputBuffer : params.inputBuffer;
+      let outputBuffer = switchBuffer ? params.inputBuffer : params.outputBuffer;
+      if (target && index === this.passes.length - 1) {
+        outputBuffer = target;
+      }
+      this.passes[index].render({inputBuffer, outputBuffer});
+      switchBuffer = !switchBuffer;
+    }
+    return {
+      inputBuffer: switchBuffer ? params.outputBuffer : params.inputBuffer,
+      outputBuffer: switchBuffer ? params.inputBuffer : params.outputBuffer
+    };
+  }
+}
+
+function normalizePasses(gl, module, id, props) {
+  if (module.filter || module.sampler) {
+    const fs = getFragmentShaderForRenderPass(module);
+    const pass = new ScreenPass(gl, {
+      id,
+      model: getModel(gl, module, fs, id, props),
+      uniforms: null
+    });
+    return [pass];
+  }
+
+  const passes = module.passes || [];
+  return passes.map(pass => {
+    const fs = getFragmentShaderForRenderPass(module, pass);
+    const idn = `${id}-${passes.length + 1}`;
```

Comment:
All passes have the same id. Do you mean to use the pass index here?

Reply:
it's a bug, will fix it.